PRO-116 Add usage telemetry page to the docs (#4025)
EventStore/EventStore: docs/usage-telemetry.md

Reviewer: Suggested edit:

**What is tracked**:
....
- Security configurations (whether it's in secure mode)
- Enabled legacy protocols: TCP client and AtomPub
cont ....

What EventStoreDB **does not** track:

- Stream names and metadata
- Event payloads and metadata
- Usernames or passwords

Reviewer: Continued:

## Disclosure (consider changing heading to  something like User Notification?)
Upon startup, EventStoreDB notifies users about the telemetry collection, both on the console and in diagnostic logs:

--

Consider editing this:
The message is not shown if telemetry collection is disabled (see below).

To this:
**Note**: If telemetry is disabled, this message won't appear.

--

Edit "When the telemetry messages are sent" to:

## Sending Frequency:
To avoid tracking short-lived instances (like containers used for tests), telemetry data is sent one hour after activation. Following this, updates are sent every 24 hours. All transmissions are logged for transparency.

--

Edit "How to opt out":
To disable telemetry, set the `EVENTSTORE_TELEMETRY_OPTOUT` environment variable to `true`.

Author: > Note: If telemetry is disabled, this message won't appear.

It does appear, that's why the text is there. It's a different message.

> after activation.

Activation is unclear. I will change it accordingly.

> To disable telemetry, 

It's impossible to disable. The text is factually correct. One can only disable transmission, but not collection.

# Usage telemetry

> Starting with version 23.10 LTS, EventStoreDB has introduced a telemetry feature. This collects anonymous, aggregated usage statistics and sends them periodically to Event Store. Telemetry data helps us refine and improve our product based on real usage patterns.

Reviewer 2: ```suggestion
Starting with version 23.10 LTS, EventStoreDB has introduced a telemetry feature. This collects anonymous, aggregated usage statistics and sends them periodically to Event Store Ltd. Telemetry data helps us refine and improve our product based on real usage patterns.
```

## What is usage telemetry

> EventStoreDB telemetry only tracks non-Personally-Identifiable Information. Collected data does not allow Event Store to fingerprint the users by any of the collected data points.

Reviewer 3: Fingeprint users _and_ company ?
we have corporate customers

Reviewer 3: I'd also add that in _no_ way we track any _data_ like stream name or events paylod & metadata

Author: It's mentioned down below.

> Examples of the telemetry data collected:

Reviewer 3: We should provide the full list of telemetry date ( as shown in the example below) 
nodeID is in the example belowe , does it allow fingerprinting and correalting that info when there is a support case ?

Author: We should not. I based the text on what others do, so it's ok. We _cloud_, but we don't have to.

Author: We can fingerprint only if someone deliberately tells us their cluster id and node id. But I don't think we'll use telemetry for support and diagnostics, we'll have different tools for that.

Reviewer 3: FYI  cluster and node IDs _are_ in the message sent to us .

    "activeChunkNumber": 2
  },
  "environment": {
    "coreCount": 24,
    "isContainer": false,
    "isKubernetes": false,
    "processorArchitecture": "X64",
    "totalDiskSpace": 1000186310656,
    "totalMemory": 68631052288
  },
  "gossip": {
    "81e9ca2a-8acc-41ed-b9d0-65a652b36801": "Leader"
  },
  "persistentSubscriptions": {
    "count": 0
  },
  "projections": {
    "customProjectionCount": 0,
    "standardProjectionCount": 5,
    "customProjectionRunningCount": 0,
    "standardProjectionRunningCount": 5,
    "totalCount": 5,
    "totalRunningCount": 5
  }
}
```

## Disclosure

On startup, EventStoreDB produces a message to the console and to the diagnostic logs, which looks similar to:

```
Telemetry
---------
EventStoreDB collects usage data in order to improve your experience. The data is anonymous and collected by Event Store Ltd.
You can opt out of sending telemetry by setting the EVENTSTORE_TELEMETRY_OPTOUT environment variable to true.
For more information visit https://eventstore.com/telemetry
```

If telemetry collection is disabled (see below), the disclosure is still shown with a different text:

```
Telemetry
---------

> EventStoreDB collects usage data in order to improve your experience. The data is anonymous and collected by EventStore Ltd.

Reviewer 3: ```suggestion
EventStoreDB collects usage data in order to improve your experience. The data is anonymous and collected by Event Store Ltd.
```

Author: It's how the message actually looks. If you want to change the message in the console output disclosure, please open a PR for that, then include change in the docs to the same PR.

You have opted out of sending telemetry by setting the EVENTSTORE_TELEMETRY_OPTOUT environment variable to true.
For more information visit https://eventstore.com/telemetry
```

## When the telemetry messages are sent

> Telemetry collection is designed to exclude short-lived instances, such as EventStoreDB containers initiated for integration tests that are subsequently terminated. Telemetry messages are dispatched one hour after a cluster or instance is activated, followed by subsequent messages every 24 hours. Each telemetry transmission is logged, and the log includes the complete telemetry message, providing full transparency into the data being collected and sent to Event Store Ltd.

Reviewer 3: Where is it logged ?
specific file or the log file configured ?

## How to opt out

> You can completely disable telemetry transmission by setting the `EVENTSTORE_TELEMETRY_OPTOUT` environment variable to `true`. EventStoreDB will still collect the information required to produce telemetry messages, and those messages will be produced to console and diagnostic log, but they will not be sent to Event Store Ltd.

Reviewer 2: ```suggestion
You can completely disable telemetry transmission by setting the `EVENTSTORE_TELEMETRY_OPTOUT` environment variable to `true`. EventStoreDB will still collect the information required to produce telemetry messages, and those messages will be produced to the console and diagnostic log, but they will not be sent to Event Store Ltd.
```

You can opt out of sending telemetry by setting the EVENTSTORE_TELEMETRY_OPTOUT environment variable to true.
For more information visit https://eventstore.com/telemetry
```

If telemetry collection is disabled (see below), the disclosure is still shown with a different text:

```
Telemetry
---------
EventStoreDB collects usage data in order to improve your experience. The data is anonymous and collected by EventStore Ltd.
You have opted out of sending telemetry by setting the EVENTSTORE_TELEMETRY_OPTOUT environment variable to true.
For more information visit https://eventstore.com/telemetry
```

## When the telemetry messages are sent

Telemetry collection is designed to exclude short-lived instances, such as EventStoreDB containers initiated for integration tests that are subsequently terminated. Telemetry messages are dispatched one hour after a cluster or instance is activated, followed by subsequent messages every 24 hours. Each telemetry transmission is logged, and the log includes the complete telemetry message, providing full transparency into the data being collected and sent to Event Store Ltd.

## How to opt out

> You can completely disable telemetry collection and transmission by setting the `EVENTSTORE_TELEMETRY_OPTOUT` environment variable to `true`.

Reviewer 2: ```suggestion
You can disable the sending of telemetry by setting the `EVENTSTORE_TELEMETRY_OPTOUT` environment variable to `true`.
```

# Usage telemetry

Starting from version 23.10 LTS, EventStoreDB includes a telemetry feature, which collects anonymous aggregated usage data, and sends it to Event Store from time to time. Telemetry data helps us to understand how our product is used by our users, so we can make the product better.

## What is usage telemetry

EventStoreDB telemetry only tracks non-Personally-Identifiable Information. Collected data does not allow Event Store to fingerprint the users by any of the collected data points.

Examples of the telemetry data collected:

* EventStoreDB version number
* Cluster or instance size
* Node machine size
* Is cluster or instance configured in secure mode
* Legacy TCP client protocol enabled
* Legacy AtomPub protocol enabled
* Number of running projections
* Number of running persistent subscriptions

What EventStoreDB **does not** track:

* Stream names
* Stream metadata
* Event payloads
* Event metadata
* Usernames and passwords

Here's an example of a full telemetry message from a single-node instance:

```json
{
  "version": "23.10.0.0",

>   "tag": "",

Author: It's not for me to decide. It's an example of the actual message sent by the current release.

>   "tag": "",

Author: If you're ready to make a change @hayley-jean - please feel free :)

>   "tag": "",

Author: Oh I got it, the tag should be there but it isn't :)

    "standardProjectionRunningCount": 5,
    "totalCount": 5,
    "totalRunningCount": 5
  }
}
```

## Disclosure

On startup, EventStoreDB produces a message to the console and to the diagnostic logs, which looks similar to:

```
Telemetry
---------
EventStoreDB collects usage data in order to improve your experience. The data is anonymous and collected by Event Store Ltd.
You can opt out of sending telemetry by setting the EVENTSTORE_TELEMETRY_OPTOUT environment variable to true.
For more information visit https://eventstore.com/telemetry
```

The message is not shown if telemetry collection is disabled (see below).

## When the telemetry messages are sent

Telemetry collection is designed to exclude short-lived instances, such as EventStoreDB containers initiated for integration tests that are subsequently terminated. Telemetry messages are dispatched one hour after a cluster or instance is activated, followed by subsequent messages every 24 hours. Each telemetry transmission is logged, and the log includes the complete telemetry message, providing full transparency into the data being collected and sent to Event Store Ltd.

## How to opt out

> You can completely disable telemetry collection and transmission by setting the `EVENTSTORE_TELEMETRY_OPTOUT` environment variable to `true`.

Reviewer 2: if they opt out, the telemetry will still be logged but it wont be sent
